scanner/src/main/java/ca/on/oicr/gsi/runscanner/scanner/processor/RevioPacBioProcessor.java

```diff
@@ -0,0 +1,413 @@
+    };
+  }
+
+  /**
+   * Extract a string expression from the metadata file and write it into the DTO.
+   *
+   * @param expression the XPath expression yielding the string
+   * @param setter writer for the string
+   * @return
+   */
+  private static RevioPacBioProcessor.ProcessMetadata processString(
+      String expression, BiConsumer<PacBioNotificationDto, String> setter) {
+    XPathExpression expr = RunProcessor.compileXPath(expression)[0];
+    return (document, dto, timeZone) -> {
+      String result = (String) expr.evaluate(document, XPathConstants.STRING);
+      if (result != null) {
+        setter.accept(dto, result);
+      }
+    };
+  }
+
+  public RevioPacBioProcessor(Builder builder, String address) {
+    super(builder);
+  }
+
+  public static RunProcessor create(Builder builder, ObjectNode parameters) {
+    JsonNode address = parameters.get("address");
+    return address.isTextual()
+        ? new RevioPacBioProcessor(builder, address.textValue().replaceAll("/+$", ""))
+        : null;
+  }
+
+  @Override
+  public Stream<File> getRunsFromRoot(File root) {
+    return Arrays.stream(
+        Objects.requireNonNull(
+            root.listFiles(f -> f.isDirectory() && RUN_DIRECTORY.matcher(f.getName()).matches())));
+  }
+
+  @Override
+  public NotificationDto process(File runDirectory, TimeZone tz) throws IOException {
+    // We create one DTO for a run, but there are going to be many wells with
+    // independent and
+    // duplicate metadata that will simply
+    // overwrite in the shared DTO. If the data differs, the last well wins.
+    PacBioNotificationDto dto = new PacBioNotificationDto();
+    dto.setPairedEndRun(false);
+    dto.setSequencerFolderPath(runDirectory.getAbsolutePath());
+
+    // Get the number of SMRT Cells in the run directory
+    int smrtCellCount =
+        (int)
+            Arrays.stream(runDirectory.listFiles())
+                .filter(
+                    cellDirectory ->
+                        cellDirectory.isDirectory()
+                            && REVIO_CELL_DIRECTORY.test(cellDirectory.getName()))
+                .count();
+    dto.setLaneCount(smrtCellCount);
+
+    // Check for presence of Transfer_Test file initially to grab start time
+    try (Stream<Path> stream = Files.walk(runDirectory.toPath())) {
+      List<Path> transferTestFiles =
+          stream
+              .filter(Files::isRegularFile)
+              .filter(file -> TRANSFER_TEST.test(String.valueOf(file.getFileName())))
+              .toList();
+
+      // Need to take the list of file paths and get the earliest creation time
+      Optional<Path> earliestCreatedFilePath =
+          transferTestFiles
+              .stream()
+              .min(Comparator.comparing(filePath -> getFileCreationTime(filePath.toFile())));
+      dto.setStartDate(
+          getFileCreationTime(
+              Objects.requireNonNull(earliestCreatedFilePath.orElse(null)).toFile()));
```

Comment:
Are transferTest files guaranteed to be present when runscanner first detects a Revio run? (Otherwise you risk an unhandled NPE here)

Reply:
It's a file. There's never any guarantee 😆

Reply:
> Removed `Objects#requireNonNull` from all places

Did this commit get pushed?